python/lsst/pipe/tasks/finalizeCharacterization.py

```diff
@@ -0,0 +1,693 @@
+        try:
+            psf_selection_result = self.make_psf_candidates.run(selected_src, exposure=exposure)
+        except Exception as e:
+            self.log.warn('Failed to make psf candidates for visit %d, detector %d: %s',
+                          visit, detector, e)
+            return None, None, measured_src
+
+        psf_cand_cat = psf_selection_result.goodStarCat
+
+        # Make list of psf candidates to send to the determiner
+        # (omitting those marked as reserved)
+        psf_determiner_list = [cand for cand, use
+                               in zip(psf_selection_result.psfCandidates,
+                                      ~psf_cand_cat['final_psf_reserved']) if use]
+        flag_key = psf_cand_cat.schema['final_psf_used'].asKey()
+        try:
+            psf, cell_set = self.psf_determiner.determinePsf(exposure,
+                                                             psf_determiner_list,
+                                                             self.metadata,
+                                                             flagKey=flag_key)
+        except Exception as e:
+            self.log.warn('Failed to determine psf for visit %d, detector %d: %s',
+                          visit, detector, e)
+            return None, None, measured_src
+
+        # At this point, we need to transfer the psf used flag from the selection
+        # catalog to the measurement catalog.
+        matched_selected, matched_measured = esutil.numpy_util.match(
+            selected_src['id'],
+            measured_src['id']
+        )
+        measured_used = np.zeros(len(measured_src), dtype=bool)
+        measured_used[matched_measured] = selected_src['final_psf_used'][matched_selected]
+        measured_src['final_psf_used'] = measured_used
+
+        # Next, we do the measurement on the psf stars ...
+        try:
+            self.measurement.run(measCat=measured_src, exposure=exposure)
+        except Exception as e:
+            self.log.warn('Failed to make measurements for visit %d, detector %d: %s',
+                          visit, detector, e)
+            return psf, None, measured_src
+
+        # And finally the ap corr map.
+        try:
+            ap_corr_map = self.measure_ap_corr.run(exposure=exposure,
+                                                   catalog=measured_src).apCorrMap
+        except Exception as e:
+            self.log.warn('Failed to compute aperture corrections for visit %d, detector %d: %s',
+                          visit, detector, e)
+            return psf, None, measured_src
+
+        try:
+            self.apply_ap_corr.run(catalog=measured_src, apCorrMap=ap_corr_map)
+        except Exception as e:
+            self.log.warn('Failed to apply aperture corrections for visit %d, detector %d: %s',
+                          visit, detector, e)
+            # Assume aperture correction map is bad, so return None.
```

Comment:
Is there a specific failure mode that we've seen that you're guarding against here?  I'm all for warning and not producing outputs when we have a known "can't handle this data" failure mode (as in the other two `try/except` blocks here), but in the absence of one I'd rather let exceptions bubble up so we're forced to deal with them more proactively the first time we see them.